analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,304 @@
+
+First, create genomic ranges objects for CNV seg file and gene.
+
+```{r}
+# Genomic range for cnv seg file 
+cnv_gr <- cnv %>%
+  # format
+  dplyr::rename(chr = chrom, start = loc.start, end = loc.end,
+                copy_number = copy.num) %>%
+  dplyr::select(-num.mark, -seg.mean) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(keep.extra.columns = TRUE,
+                                          starts.in.df.are.0based = FALSE)
+
+# Genomic range for gene location and domain overlap file 
+domain_gr <- bioMartDataPfamFGFR1 %>%
+  # keep only domain info
+  dplyr::filter(!is.na(domain_start),!is.na(domain_end)) %>%
+  # format
+  dplyr::mutate(strand = if_else(strand=="-1","-","+"),
+                domain_chr = paste0("chr",domain_chr)) %>%
+  dplyr::rename(chr = domain_chr, start = domain_start, end = domain_end) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(keep.extra.columns = TRUE,
+                                          starts.in.df.are.0based = FALSE,seqnames.field = "chr")
+
+```
+
+
+
+### Distribution of manta SV
+SV should be annotated as tandem duplicate and overlap 
+"Protein tyrosine kinase" domain of FGFR1 in biospecimens
+
+```{r}
+
+# Genomic range for manta file 
+manta_gr <- manta_lgat %>%
+  # format
+  dplyr::rename(chr = SV.chrom, start = SV.start, end = SV.end,
+                ALT = ALT) %>%
+  # format with chr to match to cnv
+  dplyr::mutate(chr=paste0("chr",chr)) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(keep.extra.columns = TRUE,
+                                          starts.in.df.are.0based = FALSE)
+
+# overlap manta and domain
+overlaps_manta <- IRanges::mergeByOverlaps(manta_gr, domain_gr)
+
+# filter for fgfr only overlap
+overlaps_manta_fgfr <- overlaps_manta[which(overlaps_manta$Gene.name=="FGFR1" &
+                # filter for tandem duplication
+                overlaps_manta$ALT=="<DUP:TANDEM>"), ]
+
+overlaps_manta_fgfr_df <- data.frame(
+  biospecimen_id = overlaps_manta_fgfr$Kids.First.Biospecimen.ID.Tumor,
+  ALT = overlaps_manta_fgfr$ALT,
+  NAME = overlaps_manta_fgfr$NAME,
+  hgnc_symbol = overlaps_manta_fgfr$hgnc_symbol,
+  # annnotation for dup tandem
+  FGFR_DUP_TANDEM = "Yes",
+  stringsAsFactors = FALSE
+) %>%
+  dplyr::distinct() 
+
+overlaps_manta_fgfr_df 
+
+```
+4 biospecimens have tandem duplication that overlap tyrosine kinase domain in FGFR1
+
+
+### Distribution of copy number
+CNV that are within FGFR1 gene region only and overlap "Protein tyrosine kinase" domain in FGFR1 in biospecimens
+
+```{r}
+
+# gene region overlap
+fgfr_gene = GenomicRanges::GRanges(seqnames=paste0("chr",domain_gr$chromosome_name),
+          ranges=IRanges::IRanges(start = domain_gr$gene_start, end = domain_gr$gene_end))
+
+# subset cnv+domain grange to be within FGFR1 gene region
+cnv_fgfr_gr <- IRanges::subsetByOverlaps(cnv_gr, fgfr_gene,type="within")
```

Comment:
Same kinds of clarification questions here since this and the next steps are important. Can you explain the goal here. You are intending to retrieve only the CNVs that are completely contained within the fgfr gene? Can you explain why you only want genes that are completely within the gene? What about if a gene is like 90% overlapped do we still want it? If so, this won't be kept so we should switch strategies.

Reply:
I believe the condition was to keep CNVs that overlap the boundaries of FGFR1 exon 10 to 18 so I've added the type="within".

Reply:
My point is about using `type = "within"` that means that: `the query interval must be wholly contained within the subject interval`https://www.rdocumentation.org/packages/GenoGAM/versions/1.0.2/topics/subsetByOverlaps%2CGenomicTiles%2CGRanges-method

Which seems pretty stringent and I'm not sure its what we want? If the concern is making sure there is enough overlap, we should instead use `maxgap`.

Reply:
I think we do want the query(CNV) to be wholly within the subject(gene) . Only with this type of overlap I would be able to get CNVs overlapping exactly/parts of the exon 10 to 18 (kinase) region, right? Am I missing something?

```r
# 1. subset cnv+domain grange to be within FGFR1 gene region
cnv_fgfr_gr <- IRanges::subsetByOverlaps(cnv_gr, fgfr_gene,type="within")

# 2. subset CNV granges within FGFR1 to only kinase domain overlapping rows
overlaps_fgfr_cnv <- IRanges::subsetByOverlaps(cnv_fgfr_gr,domain_gr) 

```

Reply:
With the "within" condition we are able to gather CNVs that exactly overlap the 10-18 exon regions.
![image](https://user-images.githubusercontent.com/34580719/99570773-720ea280-29a0-11eb-8c6f-9c6a74235737.png)

Without the "within" condition it seems we gather CNVs that span the complete gene body and not just the 10-18 exon region of interest.
![image](https://user-images.githubusercontent.com/34580719/105076283-dc06ff00-5a58-11eb-9f15-3b904cf53cf0.png)

Reply:
If you don't use within, the default is `type="any"` which means any overlap (even an overlap of 1bp) will be included. Hence why my suggestion includes to set what amount of overlap you think is reasonable (This is what I meant when I referenced `maxgap` but `minOverlap` -- they are somewhat two sides of the same coin). You would have to use the `minOverlap` argument to specify a certain amount of overlap that you think is deemed enough overlap to be certain the CNV is affecting the exon. 

But using type="within" as is right now means that 100% of base pairs in that exon must be covered by the CNV. 

What if there is one base pair in the exon not included in the CNV? To me, It seems like that exon-CNV overlap should still be considered an overlap, but using `type="within"` means that in that scenario, it would not be considered an overlap. 

This is an odd overlapping scenario to interpret `type="within"` for in my opinion because overwhelmingly the CNVs are much longer than the exons meaning that this default rule from their docs applies: `For within, it is the maximum amount by which the subject may be wider than the query.`

Note that I do not know how much scenarios like `What if there is one base pair in the exon not included in the CNV` that I mentioned above occur. But it seems worth looking into.

Reply:
Basically my fundamental question here is: How many base pairs does our CNVs and the FGFR1 exons need to overlap for us to care about it?

Reply:
hi @cansavvy and @kgaonkar6! Thanks for reviewing! What we want to do is find all instances of the tyrosine kinase domain (TKD, exons 10-18) that are duplicated in tandem (right next to the other). I do think that the first analysis @kgaonkar6 had performed gets at this information, especially after seeing the image above with the large CNVs spanning the entire gene. This would be a gene-gene duplication, rather than the smaller TKD itself duplicated).

Reply:
For the specific number of base overlap, if the CNV has 100 bp we want all 100 bp to overlap 100 bps "within" the *gene*. These tandem duplication of a group of exons will thus be smaller than the gene length, which I believe we can capture with :

```r
# 1. subset cnv+domain grange to be within FGFR1 gene region
cnv_fgfr_gr <- IRanges::subsetByOverlaps(cnv_gr, fgfr_gene,type="within")

```

Not sure how if we can specify `maxgap` beforehand, let me know what you have in mind with that parameter. Thanks!

Reply:
I am not 100% certain how the coordinates from the CNV data are obtained, but this analysis is looking at whether coordinates overlap, and not whether the base pairs sequences of the gene itself is duplicated within the CNV. So I'm not saying that this approach is wrong, but we might want to be careful about what information we can specifically say from it.

Reply:
Okay. I had to think about this a bit and go back to read a bit about the CNV data. I think what you are saying makes sense. And if we want to go with 100% overlap by using type="within" that seems fine.